core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -412,24 +436,37 @@ export class RteHelperModalComponent {
             this.ngbActiveModal.dismiss('cancel');
           }
         );
-    } else {
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
       for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
         const caName = this.tmpCustomizationArgs[i].name;
         if (caName === 'video_id') {
           const temp = this.tmpCustomizationArgs[i].value;
           customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
             temp.toString()
           );
-        } else if (caName === 'text' && this.currentRteIsLinkEditor) {
-          // Set the link `text` to the link `url` if the `text` is empty.
+        } else {
           (
             customizationArgsDict as {
               [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
                 CustomizationArgsNameAndValueArray[number]['value'];
             }
+          )[caName] = this.tmpCustomizationArgs[i].value;
+        }
+      }
+      this.ngbActiveModal.close(customizationArgsDict);
+    } else if (this.componentId === this.COMPONENT_ID_LINK) {
+      for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
+        const caName = this.tmpCustomizationArgs[i].name;
+        if (caName === 'text') {
+        // Set the link `text` to the link `url` if the `text` is empty.
+          (
+          customizationArgsDict as {
+            [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
+              CustomizationArgsNameAndValueArray[number]['value'];
+          }
           )[caName] =
-            this.tmpCustomizationArgs[i].value ||
-            this.tmpCustomizationArgs[i - 1].value;
+          this.tmpCustomizationArgs[i].value ||
```

Comment:
Please indent properly, use parens on the previous line to indicate that this is a continuation line if needed. See comments I left in previous reviews.

Reply:
done